packages/jaeger-ui-components/src/TraceTimelineViewer/SpanLinks.tsx

```diff
@@ -0,0 +1,80 @@
+import { css } from '@emotion/css';
+import React from 'react';
+import { SpanLinks } from 'src/types/links';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { useStyles2, WithContextMenu, MenuGroup, MenuItem, Icon } from '@grafana/ui';
+
+interface SpanLinksProps {
+  links: SpanLinks;
+}
+
+const renderMenuItems = (links: SpanLinks) => {
+  return (
+    <>
+      {links.logLinks ? (
+        <MenuGroup label="Logs">
+          {links.logLinks.map((link, i) => (
+            <MenuItem
+              key={i}
+              label="Logs for this span"
+              onClick={link.onClick ? link.onClick : undefined}
+              url={link.href}
+            />
+          ))}
+        </MenuGroup>
+      ) : null}
+      {links.metricLinks ? (
+        <MenuGroup label="Metrics">
+          {links.metricLinks.map((link, i) => (
+            <MenuItem
+              key={i}
+              label="Metrics for this span"
+              onClick={link.onClick ? link.onClick : undefined}
+              url={link.href}
+            />
+          ))}
+        </MenuGroup>
+      ) : null}
+      {links.traceLinks ? (
+        <MenuGroup label="Traces">
+          {links.traceLinks.map((link, i) => (
+            <MenuItem
+              key={i}
+              label={link.title ?? 'View linked span'}
```

Comment:
Definitely possible that a really long link is rendered because it can be the span name. If that happens, the menu gets rendered in the wrong place. 

I added a max-width and hide the overflow on the menu item.